checker/src/main/java/org/checkerframework/checker/index/inequality/LessThanVisitor.java

```diff
@@ -84,4 +86,27 @@ protected void commonAssignmentCheck(
         }
         super.commonAssignmentCheck(varType, valueType, valueTree, errorKey);
     }
+
+    @Override
+    protected boolean isTypeCastSafe(AnnotatedTypeMirror castType, AnnotatedTypeMirror exprType) {
+
+        AnnotationMirror exprLTAnno =
+                exprType.getEffectiveAnnotationInHierarchy(atypeFactory.UNKNOWN);
+
+        if (exprLTAnno != null) {
+            List<String> initialAnnotations =
```

Comment:
This might be `null`, see documentation of LessThanAnnotatedTypeFactory.getLessThanExpressions.